### ISSUE TITLE
Mismatch in build-type of dependencies and OpenSim can lead to runtime errors like segfault

### ISSUE THREAD

**Issue by klshrinidhi** [opensim-org/opensim-core#1276]
For example, if Simbody is built in _Release_ and OpenSim is built in _Debug_, there may be runtime errors like segfault.

If the dependencies are built using superbuild, there may be a solution in writing out some information into the install directory so that OpenSim CMake can pick it up.

**Comment by chrisdembia** [comment]
I'm hoping it is not necessary to write out additional information. All the detecting about the build configurations of the dependencies should exist, via the `SimbodyTargets-relwithdebinfo.cmake`, etc files. So I think it will just be a matter of picking up that information by querying for target properties. Perhaps there's something like `get_property(SimTKcommon PROPERTY CONFIGURATIONS)`.

**Comment by chrisdembia** [comment]
Ah the configurations in which the dependencies are built is stored in the target property `IMPORTED_CONFIGURATIONS`.

So one can use `get_property(SimTKcommon PROPERTY IMPORTED_CONFIGURATIONS)`

https://cmake.org/cmake/help/v3.0/prop_tgt/IMPORTED_CONFIGURATIONS.html

**Comment by chrisdembia** [comment]
@klshrinidhi @aymanhab There is a cmake variable for doing exactly what you want:

https://cmake.org/cmake/help/v3.0/prop_tgt/MAP_IMPORTED_CONFIG_CONFIG.html

I think all we have to do is add the following to our root CMakeLists.txt:

```cmake
set(CMAKE_MAP_IMPORTED_CONFIG_DEBUG Debug)
set(CMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO RelWithDebInfo)
set(CMAKE_MAP_IMPORTED_CONFIG_RELEASE Release)
set(CMAKE_MAP_IMPORTED_CONFIG_MINSIZEREL MinSizeRel)
```

This means that only the configs on the right can be used (from Simbody) for the configs on the left (for OpenSim's targets).

**Comment by aymanhab** [comment]
:+1:

**Comment by chrisdembia** [comment]
I discovered this via a Stack Overflow question: http://stackoverflow.com/questions/40029551/how-does-cmake-determine-the-fallback-build-configuration-for-target-dependencie/40032277#40032277

**Comment by klshrinidhi** [comment]
Addressed by #1487.
